faker/providers/person/en_NZ/__init__.py

```diff
@@ -0,0 +1,1419 @@
+        ("Rose", 1384),
+        ("Rosemary", 1918),
+        ("Ruby", 4332),
+        ("Ruth", 1616),
+        ("Sadie", 151),
+        ("Sally", 2445),
+        ("Samantha", 7549),
+        ("Sandra", 7429),
+        ("Sara", 1121),
+        ("Sarah", 19901),
+        ("Sasha", 44),
+        ("Savannah", 443),
+        ("Scarlett", 1045),
+        ("Shakira", 52),
+        ("Shania", 338),
+        ("Shannon", 2446),
+        ("Sharlene", 220),
+        ("Sharon", 7243),
+        ("Shelley", 2569),
+        ("Sheree", 169),
+        ("Sheryl", 1688),
+        ("Shirley", 1673),
+        ("Shona", 1210),
+        ("Sienna", 1358),
+        ("Sinead", 53),
+        ("Skye", 97),
+        ("Skyla", 105),
+        ("Skylar", 41),
+        ("Sofia", 630),
+        ("Sonia", 246),
+        ("Sonya", 632),
+        ("Sophia", 2595),
+        ("Sophie", 7868),
+        ("Stacey", 3037),
+        ("Stella", 1323),
+        ("Stephanie", 5794),
+        ("Summer", 1477),
+        ("Susan", 12686),
+        ("Suzanne", 4705),
+        ("Tamara", 312),
+        ("Tania", 6879),
+        ("Tanya", 1595),
+        ("Tara", 503),
+        ("Tayla", 1823),
+        ("Taylor", 1499),
+        ("Tegan", 318),
+        ("Teresa", 2294),
+        ("Tessa", 1439),
+        ("Thea", 279),
+        ("Tiana", 388),
+        ("Tina", 2124),
+        ("Toni", 2572),
+        ("Tori", 50),
+        ("Tracey", 6914),
+        ("Tracy", 3999),
+        ("Trinity", 401),
+        ("Tyla", 98),
+        ("Valerie", 394),
+        ("Vanessa", 3941),
+        ("Vicki", 3171),
+        ("Vicky", 198),
+        ("Victoria", 4823),
+        ("Violet", 506),
+        ("Virginia", 54),
+        ("Vivienne", 802),
+        ("Wendy", 6832),
+        ("Whitney", 50),
+        ("Willow", 743),
+        ("Yvonne", 1822),
+        ("Zara", 1292),
+        ("Zoe", 3973),
+        ("Zoey", 165),
+
+        ("Amaia", 667),
+        ("Ana", 730),
+        ("Anahera", 1760),
+        ("Anika", 1432),
+        ("Aria", 1960),
+        ("Ariana", 1729),
+        ("Aroha", 1796),
+        ("Ataahua", 876),
+        ("Awhina", 583),
+        ("Hana", 536),
+        ("Hinewai", 536),
+        ("Huia", 528),
+        ("Kahurangi", 730),
+        ("Kaia", 1576),
+        ("Kora", 878),
+        ("Mahi", 556),
+        ("Maia", 1960),
+        ("Manaia", 912),
+        ("Maraea", 703),
+        ("Mareikura", 948),
+        ("Mereana", 637),
+        ("Miriama", 614),
+        ("Nia", 667),
+        ("Ria", 703),
+        ("Terina", 528),
+        ("Tia", 1695),
+        ("Tiare", 671),
+        ("Tui", 1251),
+        ("Waimarie", 671),
+        ("Wikitoria", 583),
+    ))
+
+    first_names = first_names_male.copy()
+    first_names.update(first_names_female)
+
+    # New Zealand surnames compiled (and cleaned up) from the following sources:
+    #
+    # NZ Cemetery plot data:
+    #    https://catalogue.data.govt.nz/dataset?q=cemetery+plots
+    # NZ Electoral Roll (not publicly available):
```

Comment:
It turns out I didn't use that source, because it wasn't even available to me in end (it's tightly controlled by NZ legislation). I will remove the comment.